src/server/game/Scripting/ScriptMgr.h

```diff
@@ -1517,6 +1518,9 @@ class GlobalScript : public ScriptObject
 
     // Called before the phase for a WorldObject is set
     virtual void OnBeforeWorldObjectSetPhaseMask(WorldObject const* /*worldObject*/, uint32& /*oldPhaseMask*/, uint32& /*newPhaseMask*/, bool& /*useCombinedPhases*/, bool& /*update*/) { }
+
+    // Called when checking if a spell is affected by a mod
+    virtual bool OnIsAffectedBySpellModCheck(SpellInfo const* /*affectSpell*/, SpellInfo const* /*checkSpell*/, SpellModifier const* /*mod*/) { return true; };
```

Comment:
CanIsAffectedBySpellModCheck

Reply:
Hey, I appreciate the suggestion but I was told by Yehonal that hooks should follow the format: On/After/Before + <function where its used>, so thanks but I will decline this suggestion